autorest/responder.go

```diff
@@ -109,25 +100,26 @@ func ByClosingIfError() RespondDecorator {
 // ByUnmarshallingJSON returns a RespondDecorator that decodes a JSON document returned in the
 // response Body into the value pointed to by v.
 func ByUnmarshallingJSON(v interface{}) RespondDecorator {
-	return byUnmarshalling(v, "JSON", func(r io.Reader) decoder { return json.NewDecoder(r) })
+	return ByUnmarshallingAs(EncodedAsJSON, v)
 }
 
 // ByUnmarshallingXML returns a RespondDecorator that decodes a XML document returned in the
 // response Body into the value pointed to by v.
 func ByUnmarshallingXML(v interface{}) RespondDecorator {
-	return byUnmarshalling(v, "XML", func(r io.Reader) decoder { return xml.NewDecoder(r) })
+	return ByUnmarshallingAs(EncodedAsXML, v)
 }
 
-func byUnmarshalling(v interface{}, format string, df func(io.Reader) decoder) RespondDecorator {
+// ByUnmarshallingAs returns a RespondDecorator that decodes the data returned in the response
+// Body, based on the passed encoding, into the value pointed to by v.
+// encodedAs specifies the type of encoding and v provides the destination.
+func ByUnmarshallingAs(encodedAs EncodedAs, v interface{}) RespondDecorator {
```

Comment:
yeah it may be confusing to others. API services usually don't respond in other formats nowadays so I can't see anyone extending the pkg beyond json/xml.